app/controllers/api/v1/deliveries_controller.rb

```diff
@@ -147,6 +149,30 @@ def get_delivery_details
             address_attributes: address_attributes])
       end
 
+      def scheduled_date
+        scheduled_at = get_delivery_details.dig(:schedule_attributes, :scheduled_at)
+        return nil unless scheduled_at.present?
+        begin
+          Date.parse(scheduled_at)
+        rescue ArgumentError
+          nil
+        end
+      end
+
+      def validate_schedule
+        if scheduled_date.nil?
```

Comment:
Is there any possibility that schedule_date can contain a blank string? 🤔

Reply:
`.blank?` will be safer to use.

Reply:
`scheduled_date` is a method that returns `nil?` explicitely